docs/dev/table/streaming/temporal_tables.md

```diff
@@ -20,103 +20,255 @@ software distributed under the License is distributed on an
+**Note**: The grammar `METADATA FROM 'value.source.timestamp' VIRTUAL` means extract the database
+operation execution time for every changelog, it's strongly recommended defines the database operation execution time 
+as event time rather than ingestion-time or time in the record, otherwise the version extract from the changelog may
+mismatch with the version in database.
+ 
+### Defining Versioned View
 
-In Flink, this is represented by a [*Temporal Table Function*](#temporal-table-function).
+Flink also supports defining versioned view only if the view contains unique key constraint and event time.
+ 
+Let’s assume that we have the following table `RatesHistory`:
+{% highlight sql %}
+-- Define an append-only table
+CREATE TABLE RatesHistory (
+    currency_time TIMESTAMP(3),
+    currency STRING,
+    rate DECIMAL(38, 10),
+    WATERMARK FOR currency_time AS currency_time   -- defines the event time
+) WITH (
+  'connector' = 'kafka',
+  'topic' = 'rates',
+  'scan.startup.mode' = 'earliest-offset',
+  'properties.bootstrap.servers' = 'localhost:9092',
+  'format' = 'json'                                -- this is an append only source
+)
+{% endhighlight %}
 
-### Correlate with a changing dimension table
+Table `RatesHistory` represents an ever growing append-only table of currency exchange rates with respect to 
+Yen (which has a rate of 1). For example, the exchange rate for the period from 09:00 to 10:45 of Euro to Yen was 114.
+From 10:45 to 11:15 it was 116.
+
+{% highlight sql %}
+SELECT * FROM RatesHistory;
 
-On the other hand, some use cases require to join a changing dimension table which is an external database table.
+currency_time currency  rate
+============= ========= ====
+09:00:00      US Dollar 102
+09:00:00      Euro      114
+09:00:00      Yen       1
+10:45:00      Euro      116
+11:15:00      Euro      119
+11:49:00      Pounds    108
+{% endhighlight %}
 
-Let's assume that `LatestRates` is a table (e.g. stored in HBase) which is materialized with the latest rate. The `LatestRates` is the materialized history `RatesHistory`. Then the content of `LatestRates` table at time `10:58` will be:
+To define a versioned table on `RatesHistory`, Flink supports defining a versioned view 
+by [deduplication query]({{ site.baseurl }}/dev/table/sql/queries.html#deduplication) which produces an ordered changelog
+stream with an inferred primary key(`currency`) and event time(`currency_time`).
 
-{% highlight text %}
-10:58> SELECT * FROM LatestRates;
-currency   rate
-======== ======
-US Dollar   102
-Yen           1
-Euro        116
+{% highlight sql %}
+-- Define a versioned view
+CREATE VIEW versioned_rates AS              
+SELECT currency, rate, currency_time            -- (1) `currency_time` keeps the event time
+  FROM (
+      SELECT *,
+      ROW_NUMBER() OVER (PARTITION BY currency  -- (2) the inferred unique key `currency` can be a primary key
+         ORDER BY currency_time DESC) AS rowNum 
+      FROM rates )
```

Comment:
Make sure the examples are "runnable" We should use the `RatesHistory` table here.